keras/backend/cntk_backend.py

```diff
@@ -2727,3 +2727,67 @@ def forward(self, argument, device=None, outputs_to_retain=None):
 
     def backward(self, state, root_gradients):
         return root_gradients
+
+
+def reset_uids():
```

Comment:
This can be replicated in all backends.

Reply:
I believe that implementing it is outside the scope of this PR. We can leave the `NotImplementedError` and open an issue for each of those functions which need to be implemented.